Debian/Ubuntu: Modernize installation instructions

amotl opened the pull request:
### About

Modernize installation instructions for Debian/Ubuntu systems, not using the `apt-key` program any longer, because it has been deprecated. While being at it, the patch also adds the `signed-by` attribute to the package registry line.

### View

Please visit [Installation » Linux » Debian or Ubuntu](https://github.com/crate/crate-tutorials/blob/amo/improve-ubuntu-docs/docs/basic/index.rst#linux) to directly display the outcome of this change.

### Details

@proddata reported and emphasized at GH-92 what some of us may already have recognized when working on Debian- or Ubuntu-based systems.
```
Warning: apt-key is deprecated. Manage keyring files in trusted.gpg.d instead (see apt-key(8)).
```

This patch aims to fix it by adjusting the documentation accordingly, also following revised best practises for 3rd-party repositories on specifying the `signed-by` attribute to the package registry line within `/etc/apt/sources.list.d/cratedb.list`:
```
deb [signed-by=/etc/apt/trusted.gpg.d/cratedb.asc arch=amd64] https://cdn.crate.io/downloads/apt/stable/ buster main
```

I've verified the walkthrough on Docker containers with Ubuntu 20 and 22, and Debian 9, 10, and 11.

/cc @seut, @mfussenegger, @hammerhead

amotl committed:
Debian/Ubuntu: Adjust installation instructions to omit `apt-key`

Warning: apt-key is deprecated. Manage keyring files in trusted.gpg.d
instead (see apt-key(8)).

hlcianfagna commented on docs/basic/index.rst:
```diff
@@ -91,17 +91,21 @@ You will need to configure your system to register with and trust packages from
-    apt-get install sudo
-    sudo apt-get install curl gnupg software-properties-common apt-transport-https apt-utils
+    apt update
+    apt install --yes sudo
```
Do we need this step?

amotl replied:
No, it is actually an optional step, when we use `sudo` throughout the command walkthrough, and it has not been installed on the machine. Let's just remove it, and better replace the section with a single invocation `sudo apt update`, effectively expecting the program to be installed already?

hlcianfagna replied:
Without sudo you would not be able to run `apt update`

amotl replied:
I've just submitted 97c389cb7568, implementing my suggestion. WDYT?

amotl committed:
Debian/Ubuntu: Improve installation instructions to omit `sudo` setup

Let's expect the program is installed on the machine already.

hlcianfagna commented on docs/basic/index.rst:
```diff
@@ -91,27 +91,31 @@ You will need to configure your system to register with and trust packages from
 
     # Import the public GPG key for verifying the package signatures.
-    curl -sS https://cdn.crate.io/downloads/deb/DEB-GPG-KEY-crate | sudo apt-key add -
+    sudo curl -sS https://cdn.crate.io/downloads/deb/DEB-GPG-KEY-crate > /etc/apt/trusted.gpg.d/cratedb.asc
```
```suggestion
    curl -sS https://cdn.crate.io/downloads/deb/DEB-GPG-KEY-crate | sudo tee /etc/apt/trusted.gpg.d/cratedb.asc
```

amotl replied:
Thanks. Fixed with c2146b8c2b, see below.

hlcianfagna commented on docs/basic/index.rst:
```diff
@@ -91,27 +91,31 @@ You will need to configure your system to register with and trust packages from
 
+    # Register with the CrateDB package repository.
+    sudo echo "deb [signed-by=/etc/apt/trusted.gpg.d/cratedb.asc arch=amd64] https://cdn.crate.io/downloads/${repository}/stable/ ${distribution} main" \
+        > /etc/apt/sources.list.d/cratedb.list
```
```suggestion
        | sudo tee /etc/apt/sources.list.d/cratedb.list
```

hlcianfagna commented on docs/basic/index.rst:
```diff
@@ -91,27 +91,31 @@ You will need to configure your system to register with and trust packages from
-    sudo add-apt-repository "deb [arch=amd64] https://cdn.crate.io/downloads/${repository}/stable/ ${distribution} main"
 
+    # Register with the CrateDB package repository.
+    sudo echo "deb [signed-by=/etc/apt/trusted.gpg.d/cratedb.asc arch=amd64] https://cdn.crate.io/downloads/${repository}/stable/ ${distribution} main" \
```
```suggestion
    echo "deb [signed-by=/etc/apt/trusted.gpg.d/cratedb.asc arch=amd64] https://cdn.crate.io/downloads/${repository}/stable/ ${distribution} main"  \
```

amotl committed:
Debian/Ubuntu: Fix installation instructions by using `tee`

Instead of using the stdout redirection operator `>`, use the `tee`
command for writing output to a destination file, and place the `sudo`
command at the right spot.

amotl commented:
@hlcianfagna: Thank you for verifying the commands on a real machine, where I missed to catch that the operations would fail becaused I used Docker, and invoked the commands as a `root` user. c2146b8c2 fixes the instructions by using the `tee` command instead of using the stdout redirection operator `>`, and by placing the `sudo` command at the right spot.

I've verified that it didn't work before, and that it works now, by using [Racker](https://pypi.org/project/racker/). Thank you so much!
```console
# Enter sandbox.
pip install racker
racker --verbose run -it --rm ubuntu:jammy /bin/bash

# Install CrateDB from repository.
# Follow instructions at https://github.com/crate/crate-tutorials/blob/amo/improve-ubuntu-docs/docs/basic/index.rst#linux
```
